samples/net/openthread/coap_client/src/coap_client.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(coap_client, LOG_LEVEL_DBG);
+
+#include <logging/log_ctrl.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <net/socket.h>
+#include <dk_buttons_and_leds.h>
+
+#include "coap_utils.h"
+
+#define SLEEP_TIME 500
+#define IPV6_ADDR_LENGHT 16
+
+/**@brief Enumeration describing light commands. */
+typedef enum {
+	THREAD_COAP_UTILS_LIGHT_CMD_OFF = '0',
+	THREAD_COAP_UTILS_LIGHT_CMD_ON = '1',
+	THREAD_COAP_UTILS_LIGHT_CMD_TOGGLE = '2'
+} thread_coap_utils_light_command_t;
+
+static struct k_work localLightWork;
+static struct k_work meshLightWork;
+static struct k_work meshProvisioningWork;
+char peerAddr[IPV6_ADDR_LENGHT + 1] = {
+	0,
+};
+
+static int on_provisioning_reply(const struct coap_packet *response,
+				 struct coap_reply *reply,
+				 const struct sockaddr *from)
+{
+	const u8_t *pload;
+	u16_t pSize = 0u;
+
+	ARG_UNUSED(reply);
+	ARG_UNUSED(from);
+
+	pload = coap_packet_get_payload(response, &pSize);
+
+	if (pSize != IPV6_ADDR_LENGHT) {
+		LOG_ERR("Received data size is invalid");
+	}
+
+	if (!inet_ntop(AF_INET6, pload, peerAddr, IPV6_ADDR_LENGHT)) {
+		LOG_ERR("Received data is not IPv6 address: %d", errno);
+	}
+
+	LOG_INF("Received peer address: %s", log_strdup(peerAddr));
+
+	return errno;
+}
+
+static void toggle_local_light(struct k_work *item)
+{
+	static const char *const lightOption[] = { "light", NULL };
+	u8_t payload = (u8_t)THREAD_COAP_UTILS_LIGHT_CMD_TOGGLE;
+
+	ARG_UNUSED(item);
+
+	if (peerAddr[0] == 0) {
+		LOG_WRN("Peer address not set. Activate 'provisioning' option on the server side");
+		return;
+	}
+
+	LOG_INF("Send 'light' request to: %s", log_strdup(peerAddr));
+	coap_send_non_con_request(COAP_METHOD_PUT, peerAddr, lightOption,
+				  &payload, sizeof(payload));
+}
+
+static void toggle_mesh_lights(struct k_work *item)
+{
+	const char *multicastMeshLocalAddr = "ff03::1";
+	static const char *const lightOption[] = { "light", NULL };
```

Comment:
nit: Consider making those consts file static instead of function statics. Those are more like api/access definitions and should be kept together

const char *multicastMeshLocalAddr = "ff03::1";

/* some nice comment what api this is */
static const char *const lightOption[] = { "light", NULL };
static const char *const provisioningOption[] = { "provisioning",  NULL };

Reply:
done